x/sync/manager.go

```diff
@@ -63,6 +64,71 @@ const (
 	retryPriority
 )
 
+var _ DBClient = (*merkleDBSyncClient)(nil)
+
+type ClientConfig struct {
+	Hasher       merkledb.Hasher
+	BranchFactor merkledb.BranchFactor
+}
+
+type merkleDBSyncClient struct {
+	db        DB
+	config    *ClientConfig
+	tokenSize int
+	err       *utils.Atomic[error]
+	errorOnce sync.Once
+}
+
+func NewClient(db DB, config *ClientConfig) (*merkleDBSyncClient, error) {
```

Comment:
Try to avoid returning unexported types from exported code - it makes the return value incredibly awkward to use by external callers since the type isn't available to you.